src/cmdstan/command.hpp

```diff
@@ -294,29 +217,80 @@ int command(int argc, const char *argv[]) {
+    parser.print(diagnostic_csv_writers[i]);
   }
 
-  int refresh = get_arg_val<int_argument>(parser, "output", "refresh");
-
-  // arg "init" can be filename or non-negative number (init_radius)
-  std::string init = get_arg_val<string_argument>(parser, "init");
-  double init_radius = 2.0;
-  try {
-    init_radius = std::stod(init);
-    init = "";
-  } catch (const std::logic_error &e) {
-  }
-  std::vector<std::shared_ptr<stan::io::var_context>> init_contexts
-      = get_vec_var_context(init, num_chains);
-  int return_code = return_codes::NOT_OK;
-
   //////////////////////////////////////////////////
   //            Invoke Services                   //
   //////////////////////////////////////////////////
-  if (user_method->arg("generate_quantities")) {
+  int return_code = return_codes::NOT_OK;
+  std::stringstream msg;
+  auto user_method = parser.arg("method");
+
+  if (user_method->arg("pathfinder")) {
+    // ---- pathfinder start ---- //
+    auto pathfinder_arg = parser.arg("method")->arg("pathfinder");
+    int history_size
+        = get_arg_val<int_argument>(*pathfinder_arg, "history_size");
+    double init_alpha
+        = get_arg_val<real_argument>(*pathfinder_arg, "init_alpha");
+    double tol_obj = get_arg_val<real_argument>(*pathfinder_arg, "tol_obj");
+    double tol_rel_obj
+        = get_arg_val<real_argument>(*pathfinder_arg, "tol_rel_obj");
+    double tol_grad = get_arg_val<real_argument>(*pathfinder_arg, "tol_grad");
+    double tol_rel_grad
+        = get_arg_val<real_argument>(*pathfinder_arg, "tol_rel_grad");
+    double tol_param = get_arg_val<real_argument>(*pathfinder_arg, "tol_param");
+    int max_lbfgs_iters
+        = get_arg_val<int_argument>(*pathfinder_arg, "max_lbfgs_iters");
+    int num_elbo_draws
+        = get_arg_val<int_argument>(*pathfinder_arg, "num_elbo_draws");
+    int num_draws = get_arg_val<int_argument>(*pathfinder_arg, "num_draws");
+    int num_psis_draws
+        = get_arg_val<int_argument>(*pathfinder_arg, "num_psis_draws");
+    int num_paths = get_arg_val<int_argument>(*pathfinder_arg, "num_paths");
+    bool save_iterations
+        = !get_arg_val<string_argument>(parser, "output", "diagnostic_file")
+               .empty();
+    if (num_paths == 1) {
+      return_code = stan::services::pathfinder::pathfinder_lbfgs_single<
+          false, stan::model::model_base>(
+          model, *(init_contexts[0]), random_seed, id, init_radius,
+          history_size, init_alpha, tol_obj, tol_rel_obj, tol_grad,
+          tol_rel_grad, tol_param, max_lbfgs_iters, num_elbo_draws, num_draws,
+          save_iterations, refresh, interrupt, logger, init_writer,
+          sample_writers[0], diagnostic_json_writers[0]);
+    } else {
+      std::string output_file
+          = get_arg_val<string_argument>(parser, "output", "file");
+      std::string pf_name;
+      std::string pf_suffix;
+      get_basename_suffix(output_file, pf_name, pf_suffix);
```

Comment:
Instead of functions that have input/output arguments can we have this return a `std::pair<std::string, std::string>`?